stdlib/public/core/KeyPath.swift

```diff
@@ -1980,7 +2096,9 @@ func _setAtWritableKeyPath<Root, Value>(
       value: value)
   }
   // TODO: we should be able to do this more efficiently than projecting.
-  let (addr, owner) = keyPath._projectMutableAddress(from: &root)
+  let (addr, owner) = _withUnprotectedUnsafePointer(to: &root) {
```

Comment:
What is `_withUnprotectedUnsafePointer` for here?

Reply:
This is an update I'm pulling in from https://github.com/apple/swift/pull/60933.
(Rather than dealing with a formatting nightmare I downloaded KeyPath.swift from scratch and added in the offset calculations into the Walker).
Maybe I should've waited to rebase?

Reply:
Ah, I see. It might be good to rebase before we commit to make the history clean.